File: Assignment_2_User_Interactions/Metric-tracker/Metric-tracker.py
import time
from selenium import webdriver
import collections 
import csv
def writeToCSV(filename : str, metrics : dict):
    with open(file=filename, mode="w", newline="") as fp:
    # Create a writer object
        writer = csv.DictWriter(fp, fieldnames=metrics.keys())

        # Write the header row
        writer.writeheader()

        # Write the data row
        writer.writerow(metrics)
def main():

    # Initialize browser
    driver = webdriver.Chrome()

    # Navigate to your website 
    driver.get("http://localhost:3000/")
    
    #INitialize varables
    metrics = collections.defaultdict(list)
    SAMPLE_SIZE = 2
    count = 0
    start_time = time.time()

    while count < SAMPLE_SIZE:#True: #presence_time < 50: # seconds
        # Track presence time 
        # start_time = time.time()
        # presence_time = start_time
        current_time = time.time()
        presence_time = current_time - start_time
        print(f"Presence time: {presence_time} seconds")
        metrics["Presence time (Seconds)"].append(presence_time)

        # Track scrolling
        scroll_height = driver.execute_script("return document.body.scrollHeight")  
        current_scroll = driver.execute_script("return window.pageYOffset")
        print(f"Scrolled {current_scroll}/{scroll_height} pixels")
        metrics["Scrolling (PIxels)"].append(current_scroll/scroll_height)

        count += 1
        time.sleep(2) 

        # Track clicks   
        # buttons = driver.find_elements_by_tag_name("button")
        # num_clicks = 0

        # for button in buttons:
        #     button.click()
        #     num_clicks += 1
        
        # print(f"Number of clicks: {num_clicks}")
        
    driver.quit()
    print(metrics)
    writeToCSV("metrics.csv", metrics)

File: Assignment_3_Database/Metric-tracker/Metric-tracker.js

const CSV = require('csv.js');

const {By, Builder} = require('selenium-webdriver');
const assert = require("assert");

(async function firstTest() {
  let driver;
  
  // Initialize browser
  driver = await new Builder().forBrowser('Firefox').build();
  
  // Navigate to your website
  await driver.get('http://localhost:3000');
  
  //INitialize varables
  metrics = collections.defaultdict(list)
  SAMPLE_SIZE = 2
  count = 0
  var start_time = performance.now();

  while (count < sample_size) {
    
    // presence_time = start_time
    var current_time = performance.now();
    var presence_time = current_time - start_time;
    console.log("Presence time: {presence_time} miliseconds")
    metrics["Presence time (Seconds)"].append(presence_time)
    
    //const CSV = require('csv.js');
    
    //const csv = new CSV();
    //csv.parse("your,csv,data,here", function(err, data) {
    //console.log(data);
    //});
    
    // Track scrolling
    const scroll_height = driver.execute_script("return document.body.scrollHeight;")  
    const current_scroll = driver.execute_script("return window.pageYOffset;")
    console.log("Scrolled {current_scroll}/{scroll_height} pixels")
    metrics["Scrolling (PIxels)"].append(current_scroll/scroll_height)

    count += 1
    time.sleep(2) 
  }
  
  await driver.quit();
  print(metrics)
  writeToCSV("metrics.csv", metrics)

}())